File: Alpha_BackEnd_SpringBoot/src/main/java/com/jorami/eyeapp/model/Calcul.java
package com.jorami.eyeapp.model;

import com.jorami.eyeapp.model.exam.Exam;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.Where;
import org.hibernate.envers.Audited;
import org.hibernate.envers.NotAudited;

import java.util.List;


@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Audited
@SuperBuilder
@Where(clause = "deleted = FALSE")
@Entity
@Table(name="calcul")
public class Calcul extends IdentifiedModel {

    @Enumerated(EnumType.STRING)
    private com.jorami.eyeapp.model.Enum.EyeSide eyeSide;
    private Float targetRefraction;
    @ColumnDefault("false")
    private Boolean isSecondEye;
    private Float precPowerSelected;
    private Float se;
    private Long idReferencePrecExam;
    private Long idReferencePrecCalcul;

    @ManyToOne
    @JoinColumn(name="exam_id")
    private Exam exam;

    @ManyToOne
    @JoinColumn(name = "lens_id")
    private Lens lens;

    @OneToMany(cascade = CascadeType.ALL ,fetch = FetchType.LAZY, targetEntity = Constant.class, mappedBy = "calcul")
    @Fetch(FetchMode.SELECT)
    private List<Constant> constants;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, targetEntity = Diopter.class, mappedBy = "calcul")
    @Fetch(FetchMode.SELECT)
    @NotAudited
    private List<Diopter> diopters;

}

File: final_front_end_angular/src/app/body/api/calcul/Calcul.ts
import { Lens } from "../Lens";
import { Constant } from "../Constant";
import { Diopter } from "../Diopter";
import { Exam } from "../exam/Exam";

export interface Calcul {
    id?: number;
    createdBy: string;
    createdAt: string;
    modifiedBy: string;
    modifiedAt: string;
    deleted: boolean;
    version: number;

    eyeSide: 'OU' | 'OD' | 'OS' | null;
    exam: Exam | null;
    lens: Lens | null;
    constants: Constant[] | null;
    diopters: Diopter[] | null;
    targetRefraction: number | null;
    
    isSecondEye: boolean;
    precPowerSelected: number;
    se: number;
    idReferencePrecExam: number;
    idReferencePrecCalcul: number;

    constantType: string;
}
